### ISSUE TITLE
A `pulumi destroy` shouldn't be able to propose a replace

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#13098]
I ran a `pulumi destroy` today which proposed a `replace`.  This shouldn't be able to happen.

```
Previewing destroy (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/luke-pulumi-corp/demo1/dev/previews/a613d864-3df1-41fa-bd74-6f3a07ad84b3

     Type                                              Name             Plan        
 -   pulumi:pulumi:Stack                               demo1-dev        delete      
 -   ├─ docker:index:Image                             image            delete      
 -   ├─ azure-native:containerinstance:ContainerGroup  container-group  delete      
 -   ├─ azure-native:containerregistry:Registry        registry         delete      
 -   ├─ azure-native:resources:ResourceGroup           resource-group   delete      
 +-  ├─ docker:image:Image                             image            replace     
 -   └─ random:index:RandomString                      dns-name         delete      
```

Notably, the image being replaced appears to be a leftover instance from a previous attempt to replace the `image` which failed with:

```
  azure-native:containerinstance:ContainerGroup (container-group):
    error: error reading from server: read tcp 127.0.0.1:52512->127.0.0.1:52510: use of closed network connection
```

But even if this is a pending-deletion - it should not say it is going to be replaced during a `destroy`.

**Comment by dixler** [comment]
I was attempting to repro this, but was having some difficulty.

If you could DM me the state file for `luke-pulumi-corp/demo1/dev/previews/a613d864-3df1-41fa-bd74-6f3a07ad84b3`, it would be much appreciated :slightly_smiling_face:

**Comment by Frassle** [comment]
Yup this looks like the resource got tagged as for deletion, but we didn't distinguish on the second run if that tag was because it was being replaced or just deleted.

**Comment by Frassle** [comment]
Ok slightly more complicated.
That resource was marked to be replaced, but then failed it's deletion so persisted in state with that tag set.
When we run destroy we go to delete it, and see it was tagged for replacement and so report it as a delete-replace step.

But preview doesn't show replace operations separately, it treats the create side and delete side of a replace as just one "replace" operation.
In this case there is no create side, just a delete side because the create has already been done.
This would also show up on a normal update where the delete-replace had failed in the first update, the second call to "up" would delete the left over resource and say it's a delete-replace because of the tag.

I'm not sure what's best to do here. Arguably this is a display bug, but I don't think we want to split preview into showing all the different parts of replace and I'm not sure the display logic has (or wants to have) the global knowledge to see that we only issued a delete-replace and not a create-replace and change it's display based on that.

I'm going to unassign this while we think about it.
